File: IntegerBreak.py
class Solution:
    def integerBreak(self, n):
        """
        :type n: int
        :rtype: int
        """
        dp = [0, 1]
        for i in range(2, n + 1):
            m = 0
            for j in range(1, i):
                m = max(m, max(dp[j], j) * (i - j))
            dp.append(m)
        return dp[-1]


if __name__ == '__main__':
    n = int(input())
    solution = Solution()
    print(solution.integerBreak(n))

File: IntegerBreak.cpp
#include<iostream>
#include<math.h>
#include<vector>

using namespace std;

class Solution {
public:
    int integerBreak(int n) {
        vector<int> dp;
        dp.push_back(0);
        dp.push_back(1);
        for(int i = 2; i <= n; i++) {
        	int m = 0;
        	for(int j = 1; j < i; j++) {
        		m = max(m, max(dp[j], j) * (i - j));
			}
			dp.push_back(m);
		}
		
		return dp.back();
    }
};

int main() {
	
	int n;
	cin>>n;
	
	Solution *solution = new Solution();
	cout<<solution->integerBreak(n);
	
	return 0;
	
}
